### DISCUSSION TITLE
Support Jupyter syntax in regular python file with #%% marker.

### DISCUSSION THREAD

**Discussion by rowillia** [microsoft/pylance-release#4703] [Enhancement]
## Environment data

-   Language Server version: Pylance language server 2023.8.10 (pyright 932e542d)
-   OS and version: Ubuntu 22.04
-   Python version: 3.10.11

## Code Snippet

```python
# ---
# jupyter:
#   jupytex t:
#     text_representation:
#       extension: .py
#       format_name: percent
#       format_version: '1.3'
#       jupytext_version: 1.15.0
#   kernelspec:
#     display_name: Python 3
#     name: python3
# ---
#%% %autoawait trio

#%%
import trio
(await trio.run_process(
    ['echo', 'hello'], 
    capture_stdout=True)
).stdout.decode().strip()
```

## Repro Steps
Load the above file in VSCode.  Note that VSCode recognizes it as the jupytext syntax and treats it as a notebook, but still shows an error for the top level await.

## Expected behavior
No error - I would expect Jupytext notebooks to behave the same as Jupyter notebooks (this issue was fixed in https://github.com/microsoft/pylance-release/issues/1754 for Jupyter notebooks).

## Actual behavior

<img width="633" alt="Screenshot 2023-08-08 at 6 57 28 PM" src="https://github.com/microsoft/pylance-release/assets/808798/925f8d09-798d-4774-9e51-5ff386fa73b7">

**Comment by JonathanCroenen** [comment]
I am having the exact same issue right now, because top-level await in a notebook just works (and I believe with `python -m asyncio script.py` this would also work?). I would love to be able to just suppress the pyright/pylance check for this specific error in my `notebooks/` folder or something, but this error doesn't seem to have an identifier.

**Reply by rchiodo** [reply]
Pyright considers it a syntax error because in normal python code you can't have an await at the module level. There's no way to ignore syntax errors because they mess up all the parsing after that point.

**Reply by heejaechang** [reply]
we could mark the regular python file as `ipython mode` then parser will parse it as notebook cell so that won't be considered as syntax error, but not sure what kind of consequences it will cause down the road.

**Reply by JonathanCroenen** [reply]
Thanks for the reply, I understand the issue and this is ofcourse a niche usecase. You mention the ipython mode though, is that something I could manually invoke? Calling pyright with some flag to set it to ipython mode in my notebooks folder would be a simple fix!

**Reply by rchiodo** [reply]
There's an internal hack, but it's rather limited. Name the file `notebook_ipython_mode.py`

![image](https://github.com/user-attachments/assets/781e52bc-8781-4e89-883e-b19bcad3162e)

**Reply by rchiodo** [reply]
An internal team at microsoft needed this in order to support notebooks without having implemented the notebook API for lsp

**Reply by JonathanCroenen** [reply]
Haha cool, thank you. I'll have a look if I could hack something together with this.
